app/controllers/auctions_controller.rb

```diff
@@ -1,53 +1,51 @@
 class AuctionsController < ApplicationController
-  before_action :is_admin, except: [:index, :show, :bid]
-
+  before_action :is_admin, except: [:index, :show, :bid, :new]
   before_action :authenticate_user!, only: [:bid]
-
+  expose :auctions, -> { Auction.all }
+  expose :auction
+  decorates_assigned :auctions
 
   def index
     @q = Auction.ransack(search_params)
     @auctions = @q.result.page(params[:page])
   end
 
   def new
-    @auction = Auction.new
+    auction = Auction.new
   end
 
   def draw
     auction = Auction.find(params[:id])
-    users = auction.users.all
-    winner = users.order('RANDOM()').last
-    auction.winner = winner
-    auction.save
+    draw_service = DrawService.new(auction)
+    flash[:error] = draw_service.errors unless draw_service.call
     redirect_to auction
   end
 
   def create
-    @auction = Auction.new(auction_params)
-    if @auction.save
+    auction = Auction.new(auction_params)
```

Comment:
look at example controller - and try to refactor according the example 

https://github.com/hashrocket/decent_exposure#example-controller